component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1878,10 +1886,11 @@ private void publishPostSMSOTPValidatedEvent(HttpServletRequest request,
                 otpGeneratedTime);
 
         String otpExpiryTime = SMSOTPUtils.getTokenExpiryTime(context);
-        if (StringUtils.isNotEmpty(otpExpiryTime)) {
-            long expiryTime = otpGeneratedTime + Long.parseLong(otpExpiryTime);
-            eventProperties.put(IdentityEventConstants.EventProperty.OTP_EXPIRY_TIME, expiryTime);
+        if (StringUtils.isEmpty(otpExpiryTime)) {
+            otpExpiryTime = String.valueOf(SMSOTPConstants.VALIDITY_TIME);
         }
+        long expiryTime = otpGeneratedTime + Long.parseLong(otpExpiryTime);
+        eventProperties.put(IdentityEventConstants.EventProperty.OTP_EXPIRY_TIME, expiryTime);
```

Comment:
we could do something like this to avoid long -> string -> long conversion for default value.
```suggestion
        long expiryTime = otpGeneratedTime + StringUtils.isEmpty(otpExpiryTime) ? SMSOTPConstants.VALIDITY_TIME : Long.parseLong(otpExpiryTime);
        eventProperties.put(IdentityEventConstants.EventProperty.OTP_EXPIRY_TIME, expiryTime);
```